expat/configure.ac

```diff
@@ -183,33 +183,53 @@ AC_LINK_IFELSE([AC_LANG_SOURCE([
       AC_MSG_RESULT([yes])],
      [AC_MSG_RESULT([no])])])
 
+AC_ARG_WITH([getrandom],
+  [AS_HELP_STRING([--with-getrandom] [--without-getrandom],
+  [To enforce/dont use the getrandom function in the system @<:@default=check@:>@])],
+  [],
+  [with_getrandom=check])
 
-AC_MSG_CHECKING([for getrandom (Linux 3.17+, glibc 2.25+)])
-AC_LINK_IFELSE([AC_LANG_SOURCE([
-  #include <stdlib.h>  /* for NULL */
-  #include <sys/random.h>
-  int main() {
-    return getrandom(NULL, 0U, 0U);
-  }
-  ])],
-  [AC_DEFINE([HAVE_GETRANDOM], [1], [Define to 1 if you have the `getrandom' function.])
-   AC_MSG_RESULT([yes])],
-  [AC_MSG_RESULT([no])
-
-   AC_MSG_CHECKING([for syscall SYS_getrandom (Linux 3.17+)])
+AS_IF([test "x$with_getrandom" != xno],
+  [AC_MSG_CHECKING([for getrandom (Linux 3.17+, glibc 2.25+)])
+   AC_LINK_IFELSE([AC_LANG_SOURCE([
+     #include <stdlib.h>  /* for NULL */
+     #include <sys/random.h>
+     int main() {
+       return getrandom(NULL, 0U, 0U);
+     }
+     ])],
+     [AC_DEFINE([HAVE_GETRANDOM], [1], [Define to 1 if you have the `getrandom' function.])
+      AC_MSG_RESULT([yes])],
+      [AC_MSG_RESULT([no])
+       AS_IF([test "x$with_getrandom" = xyes],
+       AC_ERROR(enforced the use of sysgetrandom with --with-sysgetrandom but not detected)
```

Comment:
```console
# ./buildconf.sh
[..]
configure.ac:192: warning: The macro `AC_ERROR' is obsolete.
[..]
```
I was referring to `AC_MSG_ERROR`, not `AC_ERROR`.

Reply:
Also, the argument to `AC_MSG_ERROR` should be surrounded by `[..]`, see top of https://www.gnu.org/software/autoconf/manual/autoconf-2.67/html_node/Autoconf-Language.html .
Same for `AC_MSG_ERROR` itself, as its an argument to `AS_IF`.
And indentation of `AS_IF` arguments seems to be inconsistent with that applied to `AC_LINK_IFELSE` above. Why?

Reply:
Taken care.